src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1256,6 +1239,31 @@ static void uct_dc_mlx5_iface_handle_failure(uct_ib_iface_t *ib_iface,
     }
 }
 
+
+static int
+uct_dc_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
+                                  const uct_iface_is_reachable_params_t *params)
+{
+    uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_iface, uct_dc_mlx5_iface_t);
+    uct_dc_mlx5_iface_addr_t *addr;
+
+    if (!uct_iface_is_reachable_params_valid(
+                params, UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR)) {
+        return 0;
+    }
+
+    addr = (uct_dc_mlx5_iface_addr_t*)UCS_PARAM_VALUE(
+            UCT_IFACE_IS_REACHABLE_FIELD, params, iface_addr, IFACE_ADDR, NULL);
+    if ((addr != NULL) && !(((addr->flags & UCT_DC_MLX5_IFACE_ADDR_DC_VERS) ==
+           iface->version_flag) &&
+          (UCT_DC_MLX5_IFACE_ADDR_TM_ENABLED(addr) ==
+           UCT_RC_MLX5_TM_ENABLED(&iface->super)))) {
```

Comment:
can we simplify this expression?

Reply:
now using two levels and decomposing checks